compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -374,7 +374,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     |p| format!("`{}` defined here returns `{}`", p, callee_ty),
                                 )
                             }
-                            _ => Some(format!("`{}` defined here", callee_ty)),
+                            _ => {
+                                // If we're dealing with a local variable, its not
+                                // the struct which is defined there but the
+                                // variable is of type some struct
+                                if def.descr() == "local variable" {
+                                    Some(format!("is of type `{}`", callee_ty))
```

Comment:
Can we display the variable name too? E.g.
```
`foo` has type `u32`
```

Reply:
I think that this is the whole point of the PR. In this case (where it's a local variable), `callee_ty` is the type of variable. Right now, in the case where we are dealing with a local variable, I don't have access to the variable name which has a type of `callee_ty`. Please correct me if I'm wrong.

Reply:
When `def` is a `Res::Local`, we know it refers to the path of a variable. I would imagine it's possible to extract the name in that case (I don't have time to check right now, so let me know if you can't figure out how to do it).

Reply:
@varkor Thanks for all your comments, I've addressed 2 out of the 3. I'll work on this one.

Reply:
```suggestion
                                    Some(format!("has type `{}`", callee_ty))
```

IMHO "has type" is clearer than "is of type"